platform/ios/src/MGLMapView.mm

```diff
@@ -3759,6 +3776,32 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
     });
 }
 
+- (std::vector<MGLAnnotationTag>)shapeAnnotationTagsInRect:(CGRect)rect
+{
+    mbgl::ScreenBox screenBox = {
+        { CGRectGetMinX(rect), CGRectGetMinY(rect) },
+        { CGRectGetMaxX(rect), CGRectGetMaxY(rect) },
+    };
+    
+    std::vector<MGLAnnotationTag> nearbyAnnotations;
+    
+    mbgl::optional<std::vector<std::string>> optionalLayerIDs;
+    
+    if (_annotationTagsByAnnotation.size()) {
+        __block std::vector<std::string> layerIDs;
+        layerIDs.reserve(_annotationTagsByAnnotation.size());
+        NSString *layerID;
+        for (const auto &annotation : _annotationTagsByAnnotation) {
+            layerID = [NSString stringWithFormat:@"%@%u", MGLShapeAnnotationLayerIdentifierPrefix, annotation.second];
+            layerIDs.push_back(layerID.UTF8String);
```

Comment:
The dedicated `queryShapeAnnotations()` method should only ever return features in the shape layers; it shouldn’t be necessary for callers to build their own lists of layer IDs. Push this logic down to map.cpp, and pull [this string](https://github.com/mapbox/mapbox-gl-native/blob/453c13dee2a3678ea3d3ceda204c7cf23d431c78/src/mbgl/annotation/shape_annotation_impl.cpp#L18-L19) out into [`mbgl::AnnotationManager`](https://github.com/mapbox/mapbox-gl-native/blob/453c13dee2a3678ea3d3ceda204c7cf23d431c78/src/mbgl/annotation/annotation_manager.hpp#L47) so the code in map.cpp can reuse it.